c/auth.c

```diff
@@ -217,8 +233,21 @@ int validate_signature_eth(void *prefilled_data, const uint8_t *sig,
     }
     uint8_t out_pubkey[UNCOMPRESSED_SECP256K1_PUBKEY_SIZE];
     size_t out_pubkey_size = UNCOMPRESSED_SECP256K1_PUBKEY_SIZE;
-    ret = _recover_secp256k1_pubkey(sig, sig_len, msg, msg_len, out_pubkey,
-                                    &out_pubkey_size, false);
+
+    // https://github.com/ethereum/go-ethereum/blob/v1.13.4/crypto/signature_nocgo.go#L72
+    // The produced signature is in the [R || S || V] format where V is 0 or 1.
+    int recid = sig[RECID_INDEX];
+
+    // where the V value will be 27 or 28 for legacy reasons
+    // https://ethereum.github.io/yellowpaper/paper.pdf
+    if (recid == 27 || recid == 28) {
+        recid = recid - 27;
+    }
+    if (recid != 0 && recid != 1) {
```

Comment:
So the recid can be 0 or 1? The recovery id is among 0~3(4 values).

Reply:
I think that Ethereum should only support 0 and 1, and this value range is mentioned in the code, EIP155, and the yellow paper.